src/sinks/splunk_hec/logs/sink.rs

```diff
@@ -237,27 +233,31 @@ pub fn process_log(event: Event, data: &HecLogData) -> HecProcessedEvent {
 
     let host = data.host_key.as_ref().and_then(|key| log.get(key)).cloned();
 
-    let timestamp = data.timestamp_key.as_ref().and_then(|timestamp_key| {
-        match log.remove(timestamp_key) {
-            Some(Value::Timestamp(ts)) => {
-                // set nanos in log if valid timestamp in event and timestamp_nanos_key is configured
-                if let Some(key) = data.timestamp_nanos_key {
-                    log.try_insert(event_path!(key), ts.timestamp_subsec_nanos() % 1_000_000);
+    let timestamp = if data.endpoint_target == EndpointTarget::Event {
```

Comment:
nit: Should we `match` here to ensure that additional cases are covered if the enum is ever extended?